metadata-ingestion/src/datahub/ingestion/source/tableau/tableau_common.py

```diff
@@ -975,13 +975,23 @@ def get_filter_pages(query_filter: dict, page_size: int) -> List[dict]:
     # a few ten thousand, then tableau server responds with empty response
     # causing below error:
     # tableauserverclient.server.endpoint.exceptions.NonXMLResponseError: b''
+
+    # Keys on which pagination can be applied
+    valid_keys: List[str] = [c.ID_WITH_IN, c.PROJECT_NAME_WITH_IN]
+
+    current_key: Optional[str] = None
+
+    for key in valid_keys:
+        if key in query_filter:
+            current_key = key
+            break
+
```

Comment:
in practice, we only do pagination if `len(query_filter.keys()) == 1` 

we can move up this validation and then, we can check that

```
query_filter.keys()[0] in [c.ID_WITH_IN, c.PROJECT_NAME_WITH_IN]
```

maybe with that we can simplify the code a little bit, WDYT?

_eg_

```
if len(query_filter.keys()) != 1:
  return filter_pages

current_key = query_filter.keys()[0]
if (
  current_key in [c.ID_WITH_IN, c.PROJECT_NAME_WITH_IN]
  and query_filter.get(current_key)
  and isinstance(query_filter[current_key], list)
):
    ...

return filter_pages
```

Reply:
done